### PR TITLE
chore: remove unused clippy attributes, needless_collect

### PR DESCRIPTION
**Pull request by yihau**
#### Summary of Changes

this clippy attribute is no longer needed. remove it!

### CODE REVIEW

**Review by brooksprumo — COMMENTED** [review]
Why are these clippy annotations no longer needed?

**Comment by yihau** [comment]
maybe some false reports before 🤔  I haven't bisected it. if we're interested the root causes, I can do it

**Comment by brooksprumo** [comment]
Or, at least how was it discovered that these annotations are no longer needed?

**Comment by yihau** [comment]
we added it in bumping Rust 1.52.1: https://github.com/solana-labs/solana/pull/17096. it looks like we only run nightly clippy at that time.

**Comment by yihau** [comment]
> Or, at least how was it discovered that these annotations are no longer needed?

oh, I just tried to clean up our clippy rules and got this one haha

**Review by brooksprumo — DISMISSED** [review]

File: runtime/src/bank_forks.rs [hunk 1]
```diff
@@ -645,7 +645,6 @@ impl BankForks {
         // Clippy doesn't like separating the two collects below,
         // but we want to collect timing separately, and the 2nd requires
         // a unique borrow to self which is already borrowed by self.banks
-        #![allow(clippy::needless_collect)]
```
Comment (steviez) [inline runtime/src/bank_forks.rs:648]: The comment above can probably be tweaked. I think the comment about why we're doing two separate collects is still good, but we can probably drop / reword the "Clippy doesn't like separating the two collects below"

So, maybe just:
```
// We want to collect timing separately, and the 2nd collect requires
// a unique borrow to self which is already borrowed by self.banks
```

Comment (yihau) [reply runtime/src/bank_forks.rs:648]: great call! let me update it in a few

**Comment by anza-team** [comment]
:scream: New commits were pushed while the automerge label was present.

File: runtime/src/bank_forks.rs (hunk 1 above)
Comment (yihau) [reply runtime/src/bank_forks.rs:648]: fixed! [b119b5c](https://github.com/anza-xyz/agave/pull/1517/commits/b119b5c09ed2f77f6dcb0541766384fb164a3f25)

**Review by steviez — APPROVED** [review]

**Pull request merged**